src/components/Account/Account.tsx

```diff
@@ -262,14 +262,17 @@ const Account = ({ panelClassName, panelHeaderClassName, canUpdateEmail = true }
           formSection({
             label: t('account.terms_and_tracking'),
             saveButton: t('account.update_consents'),
-            onSubmit: (values) => updateConsents(formatConsentsFromValues(publisherConsents, { ...values.consentsValues, terms: true })),
+            onSubmit: (values) => {
+              const cleanConsentValues = Object.fromEntries(Object.entries(values.consentsValues).filter(isNotEmptyStringEntry).map(formatCrfEntry));
+              return updateConsents(formatConsentsFromValues(publisherConsents, { ...cleanConsentValues, terms: true }), cleanConsentValues);
```

Comment:
Why is `cleanConsentValues` being passed in twice to this method in different ways?

Reply:
Because cleeng requires the values to be in a `{ name: 'custom_field', version: 1, state: 'accepted', value: 'vladimir' }` format and inplayer/jwp needs them to be shaped as `{ custom_field: 'vladimir' }`. I had the formatting logic isolated inside the `updateConsents` method in `inplayer.account.service.ts` and now I have it extracted in the `isNotEmptyStringEntry` and `formatCrfEntry` functions. I can either do it this way or use those inside `updateConsents` again.